pkg/netpol/eval/internal/k8s/netpol.go

```diff
@@ -246,66 +252,108 @@ func (np *NetworkPolicy) EgressAllowedConn(dst Peer, protocol, port string) (boo
 }
 
 // GetEgressAllowedConns returns the set of allowed connetions from any captured pod to the destination peer
-func (np *NetworkPolicy) GetEgressAllowedConns(dst Peer) ConnectionSet {
+func (np *NetworkPolicy) GetEgressAllowedConns(dst Peer) (ConnectionSet, error) {
 	res := MakeConnectionSet(false)
 	for _, rule := range np.Spec.Egress {
 		rulePeers := rule.To
 		rulePorts := rule.Ports
 		peerSselected, err := np.ruleSelectsPeer(rulePeers, dst)
-		if err != nil || !peerSselected {
+		if err != nil {
+			return res, err
+		}
+		if !peerSselected {
 			continue
 		}
-		ruleConns := ruleConnections(rulePorts, dst)
+		ruleConns, err := np.ruleConnections(rulePorts, dst)
+		if err != nil {
+			return res, err
+		}
 		res.Union(ruleConns)
 		if res.AllowAll {
-			return res
+			return res, nil
 		}
 	}
-	return res
+	return res, nil
 }
 
 // GetIngressAllowedConns returns the set of allowed connections to a captured dst pod from the src peer
-func (np *NetworkPolicy) GetIngressAllowedConns(src, dst Peer) ConnectionSet {
+func (np *NetworkPolicy) GetIngressAllowedConns(src, dst Peer) (ConnectionSet, error) {
 	res := MakeConnectionSet(false)
 	for _, rule := range np.Spec.Ingress {
 		rulePeers := rule.From
 		rulePorts := rule.Ports
 		peerSselected, err := np.ruleSelectsPeer(rulePeers, src)
-		if err != nil || !peerSselected {
+		if err != nil {
+			return res, err
+		}
+		if !peerSselected {
 			continue
 		}
-		ruleConns := ruleConnections(rulePorts, dst)
+
+		ruleConns, err := np.ruleConnections(rulePorts, dst)
+		if err != nil {
+			return res, err
+		}
 		res.Union(ruleConns)
 		if res.AllowAll {
-			return res
+			return res, nil
 		}
 	}
-	return res
+	return res, nil
+}
+
+func (np *NetworkPolicy) netpolErr(title, description string) error {
+	return fmt.Errorf("network policy %s %s: %s", np.fullName(), title, description)
+}
+
+func (np *NetworkPolicy) parseNetpolCIDR(cidr string, except []string) (*IPBlock, error) {
+	ipb, err := NewIPBlock(cidr, except)
+	if err != nil {
+		return nil, np.netpolErr("CIDR error", err.Error())
```

Comment:
Maybe move "CIDR error" to be a constant like the other error titles. This way you have one place where you can see all error titles.